components/fxa-client/src/fxa_client.udl

```diff
@@ -679,6 +679,24 @@ interface FirefoxAccount {
   void simulate_permanent_auth_token_issue();
 };
 
+interface FxaStateMachine {
```

Comment:
I'm not sure what the best way forward is, but I lean slightly towards let's try to split up `FirefoxAccount` into smaller classes or something like that.  The main reason I say that is that's how the consumers are currently structured.  You normally have one module that uses a state-machine class and another module that uses `DeviceConstellation`.  That said, I think it would also be totally fine to merge it all into one class though too.

Reply:
Yeah I'm trying hard to put my "consumer" hat on, and think about what the best API to interact with is, i.e given what we know now about Android and iOS should the API be a single surface or multiple each for a specific purpose.

(I'm also totally fine exposing this from Rust, I'm mostly thinking about what Fenix/Android Components will consume and we can tie things together in a wapper if we need to)

Reply:
Talked with Tarik about this one and he convinced me that it's better to have a single interface.  Consumer apps may want to split their code into a `device-manager`, `auth-manager`, etc. but that doesn't mean we should do the same.  It's going to be annoying for consumers if we decide on a different split then they want.

TLDR: `process_event` now lives inside `FirefoxAccount` rather than a separate type.